examples/cs.py

```diff
@@ -193,4 +209,74 @@ def dcs(w):
     ionization_osi = ionization_grp.create_dataset("outer_shells", data=s.elf_file.get_outer_shells().to('eV'))
     ionization_osi.attrs['units'] = 'eV'
 
+    # electron range
+    e_ran = np.logspace(-2, 3, 129) * units.eV
+    p_ran = np.linspace(0.0, 1.0, 1024)
+
+    electron_range_grp = outfile.create_group("electron_range")
+    ran_energies = electron_range_grp.create_dataset("energy", data=e_ran.to('eV'))
+    ran_energies.attrs['units'] = 'eV'
+    ran_range = electron_range_grp.create_dataset("range", e_ran.shape)
+    ran_range.attrs['units'] = 'm'
+
+    ran_tmfp = np.zeros(e_ran.shape) * units.m;
+    tmfpmax = 0. * units.m
+    print("# Computing transport mean free path.")
+    for i, K in enumerate(e_ran):
+        def dcs(theta):
+            return elastic_cs_fn(theta, K) * (1 - np.cos(theta)) * s.rho_n
+        inv_tmfp = compute_elastic_tcs(dcs, p_ran)
+        tmfp = 1./inv_tmfp
+        if ( K >= s.band_structure.barrier):
+            # make sure the tmfp is monotonously increasing for energies above
+            # the vacuum potential barrier
+            if (tmfp < tmfpmax):
```

Comment:
Suggestion: `tmfpmax = np.maximum(tmfp_max, tmfp)` instead of these four lines and change line 239 to `ran_tmfp[i] = tmfp_max`.

So do not change the `tmfp` variable, conceptually it will always hold the tmfp for the current kinetic energy. Meanwhile, `tmfp_max` is the largest one you have found (bar anything below the vacuum barrier), and you make it clear that that's what you want to write to `ran_tmfp`.